src/fmu/sumo/explorer/objects/_search_context.py

```diff
@@ -1502,6 +1502,16 @@ def __verify_aggregation_operation(
             "key"
         ]
         classname = sres["aggregations"]["class"]["buckets"][0]["key"]
+
```

Comment:
I'm not sure if a filter on realizations is necvessary found on the top level of the generated query. A more robust check would be to create a new, temporary `SearchContext` using just the entity and the ensemble name, and verify that the length (number of hits) of this `SearchContext` is the same as the current one. Something like

```
sc=SearchContext().filter(entity=entityuuid, ensemble=ensemblename)
if len(sc) != tot_hits:
  raise Exception(...)
```

Reply:
Yes, it would, but it might not necessarily be on the top level of the query... it might somewhere deep into a nested query (not sure if this is actually possible at the moment, but at some point in the future, we might build a mechanism for creating advanced sub-queries).